go/libraries/doltcore/env/remotes.go

```diff
@@ -45,6 +45,7 @@ var ErrCannotPushRef = errors.New("cannot push ref")
 var ErrNoRefSpecForRemote = errors.New("no refspec for remote")
 var ErrInvalidSetUpstreamArgs = errors.New("invalid set-upstream arguments")
 var ErrInvalidFetchSpec = errors.New("invalid fetch spec")
+var ErrPullOnlyNoUpstream = errors.New("There is no tracking information for the current branch.\nPlease specify which branch you want to merge with.")
```

Comment:
How about naming this one `ErrPullWithNoRemoteAndNoUpstream`? I think that would be a little clearer. 

Also, it would be awesome to include the "next steps" users may need to take, like git is so good at doing. I included the git output below as an example. It would be great to tell people they can run git pull remote branch to make their pull work, or that they can set the upstream (we still need to implement dolt branch --set-upstream though, right?)
```
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> branch3
```

Reply:
Right! I'll add this with `dolt push --set-upstream <remote> <branch>` option instead of `branch --set-upstream` for now.